packages/synapse-react-client/src/synapse-queries/team/useTeamMembers.ts

```diff
@@ -114,11 +145,49 @@ export function useAddMemberToTeam() {
         queryClient.invalidateQueries({
           queryKey: keyFactory.getTeamMembersQueryKey(variables.teamId),
         }),
+        queryClient.invalidateQueries({
+          queryKey: keyFactory.getAllSubmissionTeamsQueryKeys(),
+        }),
```

Comment:
joining a team may affect whether you belong to a submission team, so invalidate that data in the cache

Reply:
It's up to the developer to know what possible service calls would be impacted by a call today, which leaves us vulnerable to stale data bugs across the library.  I don't think there's anything we can do about it (unless we had a graph of all of the API calls and links/interactions).